From 09b6e78d1592ce10fdc975025d699ee41444aa3f Mon Sep 17 00:00:00 2001
From: Paul Eggert <eggert@cs.ucla.edu>
Date: Fri, 5 Feb 2016 21:06:20 -0800
Subject: [PATCH] Fix memory leak in AC_FUNC_MMAP

* lib/autoconf/functions.m4 (AC_FUNC_MMAP): Fix memory leak
in test case, found by configuring with gcc -fsanitize=address.
---
 lib/autoconf/functions.m4 | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/autoconf/functions.m4 b/lib/autoconf/functions.m4
index 5d55e2db5087..51ae5a42ca33 100644
--- a/lib/autoconf/functions.m4
+++ b/lib/autoconf/functions.m4
@@ -1342,6 +1342,8 @@ main (void)
     if (*(data + i) != *(data3 + i))
       return 14;
   close (fd);
+  free (data);
+  free (data3);
   return 0;
 }]])],
 	       [ac_cv_func_mmap_fixed_mapped=yes],
-- 
2.10.2

